Added user guide for `geometry/Cal3_S2`

CodeXTL opened the pull request:
Main Update(s):
- Made several updates to the user guide for in order to improve overall quality of user guide.
- Added an additional blurb to the beginning to highlight the key purpose of `Cal3_S2`.
- Added more math to improve comprehension.
- Updated "Properties" section to emphasize what each parameter of the model means.
- Fix "Calibration Operations" section. In the initial draft, explanation of `calibrate()` and `uncalibrate` were switched. It has now been corrected and made clearer.
- Added "Additional Resources" section to curated resources providing foundational theoretical knowledge related to the topic of the user guide.

Minor Update(s):
- Corrected spelling in `Cal3_S2.h`

Needs Review:
- My understanding of manifolds is still lacking, so the "Manifold Operations" section needs to be reviewed. It will likely be updated in the future after I learn more about manifolds.

CodeXTL committed:
Added initial draft of Cal3_S2 user guide

CodeXTL committed:
Merge branch 'borglab:develop' into docs/geometry/Cal3_S2

CodeXTL committed:
Made large quality updates to Cal3_S2 user guide

dellaert reviewed (commented):
Discussed some issues in meeting, but looking great already!

CodeXTL committed:
Improved Cal3_S2 user guide

CodeXTL committed:
docs: clarify Cal3_S2 manifold description

CodeXTL committed:
Merge pull request #1 from CodeXTL/codex/edit-cal3_s2.ipynb-markdown-cell

Update manifold section in Cal3_S2 docs

CodeXTL committed:
Minor grammatical/syntax fix

CodeXTL commented:
Main Update(s):
- Changed from "a simple 5-parameter camera calibration model" to "the simple 5-parameter camera calibration model".
- Fixed description of `principalPoint()` member function under the "Properties" section.
- Split "Properties" into two sub-sections: "Main Parameters" and "Derived Properties".
- `calibrate()` and `uncalibrate` each has their own sub-sections under "Basic Operations".
- Mentioned how operations like `retract` and `localCoordinates` enable gradient-based optimization.

Minor Update(s):
- Spelling and grammar mistakes

dellaert reviewed (approved):
Awesome. Two more notes but can be folded in your next PR
- DOI references 
- Why is it a manifold: to optimize over it.
